src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -537,56 +537,58 @@ class HealthCertificateService {
 	func updateValidityStatesAndNotifications(shouldScheduleTimer: Bool = true) {
 		Log.info("Update validity state and notifications.")
 
-		let currentAppConfiguration = appConfiguration.currentAppConfig.value
-		healthCertifiedPersons.forEach { healthCertifiedPerson in
-			healthCertifiedPerson.healthCertificates.forEach { healthCertificate in
-				let expirationThresholdInDays = currentAppConfiguration.dgcParameters.expirationThresholdInDays
-				let expiringSoonDate = Calendar.current.date(
-					byAdding: .day,
-					value: -Int(expirationThresholdInDays),
-					to: healthCertificate.expirationDate
-				)
+		DispatchQueue.global(qos: .default).async {
+			let currentAppConfiguration = self.appConfiguration.currentAppConfig.value
+			self.healthCertifiedPersons.forEach { healthCertifiedPerson in
+				healthCertifiedPerson.healthCertificates.forEach { healthCertificate in
+					let expirationThresholdInDays = currentAppConfiguration.dgcParameters.expirationThresholdInDays
+					let expiringSoonDate = Calendar.current.date(
+						byAdding: .day,
+						value: -Int(expirationThresholdInDays),
+						to: healthCertificate.expirationDate
+					)
 
-				let previousValidityState = healthCertificate.validityState
+					let previousValidityState = healthCertificate.validityState
 
-				let blockedIdentifierChunks = appConfiguration.currentAppConfig.value
-					.dgcParameters.blockListParameters.blockedUvciChunks
-				if healthCertificate.isBlocked(by: blockedIdentifierChunks) {
-					healthCertificate.validityState = .blocked
-				} else {
-					let signatureVerificationResult = self.dccSignatureVerifier.verify(
-						certificate: healthCertificate.base45,
-						with: self.dscListProvider.signingCertificates.value,
-						and: Date()
-					)
+					let blockedIdentifierChunks = self.appConfiguration.currentAppConfig.value
```

Comment:
this might leak memory

Reply:
This comment is also in regard to the strong self reference, right?

Reply:
👍